app/src/main/java/com/example/practice/MainViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package com.example.practice
+
+import android.util.Log
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import api.*
+import com.example.practice.model.UserModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import retrofit2.Call
+import retrofit2.Response
+import retrofit2.converter.gson.GsonConverterFactory
+
+import retrofit2.Retrofit
+
+
+
+
+class MainViewModel( ): ViewModel(),API{
+    val TAG : String = this.javaClass.simpleName
+    private var usersList: MutableLiveData<List<UserModel>>? = null
+
+    //we will call this method to get the data
```

Comment:
The comment mentions this getter is used to get the data. However, this getter is also used to **set** the data  at line 39. This means, the same data is updated from both the view model and the activity class. 
Can this be improved? (You don't need to fix this but try to discuss below why updating the same data from multiple places should be avoided and how it can be avoided in this case)